llvm/lib/Target/AIE/AIE2PreLegalizerCombiner.cpp

```diff
@@ -288,6 +292,89 @@ bool AIE2PreLegalizerCombinerImpl::tryToCombineVectorInserts(
   return true;
 }
 
+/// Determines if it is safe to combine vextract by checking the uses of DstReg,
+/// specfically for a pattern involving TRUNC followed by EXT.
+bool AIE2PreLegalizerCombinerImpl::canCombineVExtractElt(
+    Register DstReg, bool SignVal, unsigned SrcEltSize) const {
+  // Returns the single non-debug use of a register with a specific opcode
+  // and destination size.
+  auto GetOneUseWithOpCode =
+      [&](const Register Reg, const unsigned OpcodeToCheck,
+          const unsigned DstSize) -> std::optional<MachineInstr *> {
+    if (MRI.hasOneNonDBGUser(Reg)) {
+      MachineInstr &Use = *MRI.use_nodbg_instructions(Reg).begin();
+      if (Use.getOpcode() == OpcodeToCheck) {
+        const LLT DstRegTy = MRI.getType(Use.getOperand(0).getReg());
+        if (DstRegTy.getSizeInBits() == DstSize)
+          return &Use;
+      }
+    }
+    return std::nullopt;
+  };
+
+  if (auto Trunc =
+          GetOneUseWithOpCode(DstReg, TargetOpcode::G_TRUNC, SrcEltSize)) {
+    MachineInstr *TruncMI = Trunc.value();
+    const unsigned ExtOpcode =
+        SignVal ? TargetOpcode::G_SEXT : TargetOpcode::G_ZEXT;
+    const Register UseDstReg = TruncMI->getOperand(0).getReg();
+    if (auto Ext = GetOneUseWithOpCode(UseDstReg, ExtOpcode, 20)) {
```

Comment:
nit: we can use:
```
    if (GetOneUseWithOpcode(UseDstReg, ExtOpcode, 20)) {
      return true;
    }
```

Because there is no further use for `auto Ext`.